crates/solidity/testing/snapshots/cst_output/ContractDefinition/unterminated_body/generated/0.4.11.yml

```diff
@@ -5,12 +5,12 @@ Source: >
 
 Errors: # 1 total
   - >
-    Error: Expected '/' or 'a' or 'b' or 'c' or 'e' or 'f' or 'i' or 'm' or 'p' or 'r' or 's' or 't' or 'u' or '}'.
+    Error: Expected Identifier, or 'mapping', or 'function', or UnsignedFixedType, or SignedFixedType, or UnsignedIntegerType, or SignedIntegerType, or FixedBytesType, or 'payable', or 'address', or 'bool', or 'string', or 'error', or 'event', or 'type', or 'enum', or 'struct', or 'modifier', or 'receive', or 'fallback', or 'function', or 'constructor', or 'using', or '}'
```

Comment:
nit: should we have a limit on how many alternatives we generate?

Reply:
This will have to wait. We need to think about the purpose of the message. We cannot arbitrarily limit the number, we would have to replace it with `ContractBodyElement` and a hyperlink to the spec. Otherwise there's no point in listing alternatives. And ordering is a matter for heuristics i.e. which is more likely the error. Currently it is determined by the sequence of alternatives in the production, which cannot always be rearranged due to the ordered choice parsing strategy.

Reply:
Sounds good 👍